src/Build/BackEnd/Client/MSBuildClient.cs

```diff
@@ -354,14 +362,11 @@ private ServerNodeHandshake GetHandshake()
         /// </summary>
         private void HandleCancellation()
         {
-            // TODO.
-            // Send cancellation command to server.
-            // SendCancelCommand(_nodeStream);
+            TrySendCancelCommand();
 
             Console.WriteLine("MSBuild client cancelled.");
             CommunicationsUtilities.Trace("MSBuild client cancelled.");
```

Comment:
I thought a bit about two lines above and I have some concerns about race conditions for client and server here . Here are situations that I worry about:
1. We cancel the build and show these lines. The server though is still sending the console output and it is received and showed by the client. Result: these lines go up and are not visible. Could this happen and what do we think about that situation?
2. We cancel the build and show this line, but the server is already finished and all the output now is in the message queue. We indeed would show the success in the console, because server thinks build is already finished, right? But the client status would be "Cancelled".

Reply:
We could fix this problem and my concern if we first check the return type to verify that the cancel command was sent successfully, then wait for a "cancelled" response from the server before continuing.

Reply:
I'd recommend to keep `CommunicationsUtilities.Trace` and delete `Console.WriteLine("MSBuild client cancelled.");`
1st of all this text shall be localized
2nd Lets hope that redirected output from server will contain enough information for customers to undestand that build has been cancelled.

Reply:
Localized message about aborting is emitted from the handler. I removed the WriteLine from this method based on our conversation.